src/language-js/printer-estree.js

```diff
@@ -3888,60 +3895,48 @@ function printTypeAnnotation(path, options, print) {
   ]);
 }
 
-function printFunctionParams(path, print, options, expandArg, printTypeParams) {
+function printFunctionParameters(
+  path,
+  print,
+  options,
+  expandArg,
+  printTypeParams
+) {
   const fun = path.getValue();
   const parent = path.getParentNode();
-  const paramsField = fun.parameters ? "parameters" : "params";
   const isParametersInTestCall = isTestCall(parent);
   const shouldHugParameters = shouldHugArguments(fun);
 
   const typeParams = printTypeParams
     ? printFunctionTypeParameters(path, options, print)
     : "";
 
-  const printedParameters = [];
-  if (fun.this) {
-    printedParameters.push({ node: fun.this, doc: path.call(print, "this") });
-  }
-
-  if (fun[paramsField]) {
-    path.each((childPath) => {
-      printedParameters.push({
-        node: childPath.getValue(),
-        doc: print(childPath),
-      });
-    }, paramsField);
-  }
+  const parameters = getFunctionParameters(fun);
+  const printedParameters = iterateFunctionParametersPath(path, print);
```

Comment:
This function doesn't return an array now. If it's going to return it, then I think it's better to call it `map` than `iterate`.

Reply:
I was going to make it not return, but found only use in this place, so already renamed.